src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java

```diff
@@ -621,18 +621,21 @@ public Void visitAttribute(AttributeTree tree, Void ignore) {
             }
 
             if (attr != null) {
+                Pattern validPattern = null;
                 switch (attr) {
                     case NAME:
                         if (currTag != HtmlTag.A) {
                             break;
                         }
+                        validPattern = validName;
                         // fallthrough
                     case ID:
                         String value = getAttrValue(tree);
+                        validPattern = validPattern != null ? validPattern : validId;
```

Comment:
This is wrong/weak.  If nothing else, this is version-specific.   For HTML 4, name and id are restricted, (`validName`) whereas for HTML5, name is not allowed, and id is almost unrestricted (`validId`)

Reply:
Yes, this change presumes 8247957.  Would this be better to merged into 8247957, then closed as duplicate?
Otherwise, I'll use env.htmlVersion here.

Reply:
I think it would be better to move forward on the bigger picture of removing support fo HTML 4, in which case you could handle this issue as well.    But if you do that, the code here should just reject use of the name attribute outright, and not attempt to validate the value according to the old rules, and the id attribute should be unconditionally checked against the HTML 5 rules.